crates/noirc_frontend/src/monomorphization/mod.rs

```diff
@@ -746,15 +768,51 @@ impl<'interner> Monomorphizer<'interner> {
         id: node_interner::ExprId,
     ) -> ast::Expression {
         let func = Box::new(self.expr(call.func));
-        let arguments = vecmap(&call.arguments, |id| self.expr(*id));
+        let mut arguments = vecmap(&call.arguments, |id| self.expr(*id));
+        let hir_arguments = vecmap(&call.arguments, |id| self.interner.expression(id));
         let return_type = self.interner.id_type(id);
-        let return_type = Self::convert_type(&return_type);
+        let return_type = self.convert_type(&return_type);
         let location = call.location;
 
+        if let ast::Expression::Ident(ident) = func.as_ref() {
+            if let Definition::Oracle(name) = &ident.definition {
+                if name.as_str() == "println" {
+                    self.append_abi_arg(&hir_arguments[0], &mut arguments);
```

Comment:
I can't seem to wrap my head around how this works. It looks like we're checking if the definition of the function being called is println. I don't see how that'd work for functions passed in as arguments but after testing it seems to handle this case just fine.

Also inside append_abi_arg we match on the first argument and always expect it to be an identifier. Yet we can still call e.g. `println(Foo { a, b: 0 })` and have it work.

I think I'm missing something or just reading this code wrong, can you explain what it is doing?

Reply:
Because the `println` oracle is contained in a wrapper:
```
unconstrained fn println<T>(input: T) {
    println_oracle(input);
}
```
it will always be an ident even when you call `println(Foo { a, b: 0 })` or pass a function as an argument

Reply:
Can we add this as a comment in the noir stdlib code as well? Otherwise someone seeing that alone may be tempted to remove the middleman and rename the oracle to println.

Reply:
I put the comment about the `println_oracle` definition in the stdlib